### PR TITLE
Implement `process_call` for LinearizeTrace

### PR DESCRIPTION
**Pull request by dougalm**

### CODE REVIEW

File: jax/_src/interpreters/ad.py [hunk 1]
```diff
@@ -82,6 +82,26 @@ def jvpfun(f, instantiate, transform_stack, primals, tangents):
...
+    tracers = [LinearizeTracer(linearize_trace, p, t) for p, t in zip(primals, tangents)]
+    with core.set_current_trace(linearize_trace):
+      ans = _f(*tracers)
+  out_primals, out_tangents = unzip2(map(linearize_trace.to_primal_tangent_pair, ans))
+  nzs_out = [type(t) is not Zero for t in out_tangents]
+  out_tangents = [t for t, nz in zip(out_tangents, nzs_out) if nz]
+  out_tangents = map(tangent_trace.to_jaxpr_tracer, out_tangents)
+  jaxpr, consts, attrs_tracked = tangent_trace.to_jaxpr(out_tangents)
+  num_residuals = len(consts)
+  del attrs_tracked # TODO: attrs
```
Comment (mattjj) [inline jax/_src/interpreters/ad.py:101]: Raise an exception if there are any? (You can also unpack into an empty tuple to check the same, though with no explanation of what the error means.)

File: jax/_src/lax/lax.py [hunk 2]
```diff
@@ -4153,9 +4153,10 @@ def _broadcast_in_dim_typecheck_rule(
 
 def _broadcast_in_dim_transpose_rule(ct, operand, *dyn_shape,
                                      shape, broadcast_dimensions, sharding):
+  aval = core.get_aval(operand)
```
Comment (mattjj) [inline jax/_src/lax/lax.py:4156]: Huh, I would've thought operand would be an UndefinedPrimal here. Does get_aval work on that? Why this change?

**Review by mattjj — APPROVED** [review]

File: jax/_src/lax/lax.py (hunk 2 above)
Comment (dougalm) [reply jax/_src/lax/lax.py:4156]: Good point. I changed it because `operand` was sometimes a literal. I thought the implementation was just missing that case (sometimes we assume things are tracers when they can be literals or tracers). But you're right, something else must be going wrong. I'll investigate.

Comment (dougalm) [reply jax/_src/lax/lax.py]: Figured it out. I was instantiating zeros inside the linear jaxpr rather than outside it. Gotta dedent your instantiate_zeros.
